spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -153,20 +164,21 @@ else if (messageListener instanceof MessageListener) {
 			throw new IllegalStateException("messageListener must be 'MessageListener' "
 					+ "or 'AcknowledgingMessageListener', not " + messageListener.getClass().getName());
 		}
-		if (getContainerProperties().getConsumerTaskExecutor() == null) {
+		if (containerProperties.getConsumerTaskExecutor() == null) {
 			SimpleAsyncTaskExecutor consumerExecutor = new SimpleAsyncTaskExecutor(
 					(getBeanName() == null ? "" : getBeanName()) + "-kafka-consumer-");
-			getContainerProperties().setConsumerTaskExecutor(consumerExecutor);
+			containerProperties.setConsumerTaskExecutor(consumerExecutor);
 		}
-		if (getContainerProperties().getListenerTaskExecutor() == null) {
+		if (containerProperties.getListenerTaskExecutor() == null) {
 			SimpleAsyncTaskExecutor listenerExecutor = new SimpleAsyncTaskExecutor(
 					(getBeanName() == null ? "" : getBeanName()) + "-kafka-listener-");
-			getContainerProperties().setListenerTaskExecutor(listenerExecutor);
+			containerProperties.setListenerTaskExecutor(listenerExecutor);
 		}
 		this.listenerConsumer = new ListenerConsumer(this.listener, this.acknowledgingMessageListener);
-		this.listenerConsumerFuture = getContainerProperties()
+		this.listenerConsumerFuture = containerProperties
 				.getConsumerTaskExecutor()
 				.submitListenable(this.listenerConsumer);
+		setRunning(true);
```

Comment:
We should place this before submitting the tasks, otherwise there's a race condition that makes it possible for the listener consumer to exit immediately.